crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1089 @@
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.flare))
+/// ```
+///
+/// ### Crest
+///
+/// The [`crest`]($gradient.crest) gradient is a blue to white to red gradient
+/// that is perceptually uniform. Crest does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.crest))
+/// ```
+///
+/// ### Regarding presets like "jet" and "parula"
+///
+/// - [Jet](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/)
+///   is not a good color map, as it is not perceptually uniform. As such,
+///   it is not color blind friendly and should not be used for data visualization,
+///   due to which it is not included in Typst.
+/// - [Parula](https://www.mathworks.com/help/matlab/ref/parula.html)
+///   is a good color map included in MATLAB, but it is not included as
+///   a preset in Typst. This is because it is owned by MathWorks and is not public.
+///
+#[ty(scope)]
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Gradient {
+    Linear(LinearGradient),
+}
+
+impl Debug for Gradient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Linear(linear) => linear.fmt(f),
+        }
+    }
+}
+
+#[scope]
+impl Gradient {
+    pub const SPECTRAL: fn() -> Array = spectral;
+    pub const VIRIDIS: fn() -> Array = viridis;
+    pub const INFERNO: fn() -> Array = inferno;
+    pub const MAGMA: fn() -> Array = magma;
+    pub const PLASMA: fn() -> Array = plasma;
+    pub const ROCKET: fn() -> Array = rocket;
+    pub const MAKO: fn() -> Array = mako;
+    pub const VLAG: fn() -> Array = vlag;
+    pub const ICEFIRE: fn() -> Array = icefire;
+    pub const FLARE: fn() -> Array = flare;
+    pub const CREST: fn() -> Array = crest;
+
+    /// Creates a new linear gradient.
+    #[func(title = "Linear gradient")]
+    pub fn linear(
+        /// The call site of this function.
+        span: Span,
+
+        /// The color stops of the gradient.
+        #[variadic]
+        stops: Vec<Spanned<Stop>>,
+
+        /// The direction or angle of the gradient.
+        #[named]
+        #[default(DirOrAngle::Dir(Dir::LTR))]
+        dir: DirOrAngle,
+
+        /// The color space in which to interpolate the gradient.
+        ///
+        /// Defaults to a perceptually uniform color space called
+        /// [Oklab]($color.oklab).
+        #[named]
+        #[default(ColorSpace::Oklab)]
+        space: ColorSpace,
+
+        /// The relative placement of the gradient.
+        ///
+        /// - `"this"`: The gradient is relative to the bounding box of the
+        ///   container onto which it is painted.
+        /// - `"parent"`: The gradient is relative to the bounding box of the
+        ///   parent that contains the element onto which the gradient is applied.
```

Comment:
If you derive `Cast` on `Relative` and add the comments as doc comments of the enum variants, the docs would automatically render this correctly, so that you don't need to add this list here.